File: src/AuthApp.Domain/Permissions/PermissionCatalog.cs
namespace AuthApp.Domain.Permissions;

public static class PermissionCatalog
{
    public const string ManageUsers = "permissions.manage-users";
    public const string ViewUsers = "permissions.view-users";
    public const string ManagePermissions = "permissions.manage-permissions";
    public const string ViewPermissions = "permissions.view-permissions";

    public static IReadOnlyCollection<string> All => new[]
    {
        ManageUsers,
        ViewUsers,
        ManagePermissions,
        ViewPermissions
    };
}

File: frontend/src/utils/permissionCatalog.ts
export const PermissionCatalog = {
  ManageUsers: 'permissions.manage-users',
  ViewUsers: 'permissions.view-users',
  ManagePermissions: 'permissions.manage-permissions',
  ViewPermissions: 'permissions.view-permissions',
} as const;

export type PermissionKey = (typeof PermissionCatalog)[keyof typeof PermissionCatalog];
